File: src/pattern/adapter/java/PrintBanner.java
package pattern.adapter.java;

public class PrintBanner extends Banner implements Print{
    public PrintBanner(String name){
        super(name);
    }

    @Override
    public void printStrong() {
        showWithAster();
    }

    @Override
    public void printWeak() {
        showWithParen();
    }
}

File: src/pattern/adapter/kotlin/PrintBanner.kt
package pattern.adapter.kotlin

class PrintBanner: Banner, Print{
    constructor(name: String): super(name)
    constructor(name: String, phone: String): super(name, phone){
        println("execute secondary constructor")

    }

    override fun printStrong() {
        showWithAster()
    }

    override fun printWeak() {
        showWithParen()
    }


}